ministryofjustice/correspondence_tool_staff
# Ct 3888 solicitor address book search

## Description
Adds the ability for users to search for solicitor addresses and default relationship type on requestor details page.



## Self-review checklist



### Screenshots
#### Before

<img width="808" alt="Screenshot 2021-10-21 at 14 27 51" src="https://user-images.githubusercontent.com/13377553/138287521-89413f42-dfc1-48b2-abe9-afb78571c25a.png">

#### After

##### Address search
<img width="791" alt="Screenshot 2021-10-21 at 14 19 45" src="https://user-images.githubusercontent.com/13377553/138287148-bc0bf472-c4f1-4cde-8310-090447fb4f75.png">

##### Default to solicitor
<img width="796" alt="Screenshot 2021-10-21 at 14 18 28" src="https://user-images.githubusercontent.com/13377553/138287233-4e6f8bba-72dd-4b31-a143-a9cf92a72b3a.png">

##### Select other type
<img width="763" alt="Screenshot 2021-10-21 at 14 19 23" src="https://user-images.githubusercontent.com/13377553/138287304-67b8dcd2-68ee-4d68-bf4b-5f268228d849.png">




### Related JIRA tickets
Part of: [CT-3814](https://dsdmoj.atlassian.net/browse/CT-3814)

Covers specifically: [CT-3888](https://dsdmoj.atlassian.net/browse/CT-3888) and [CT-3889](https://dsdmoj.atlassian.net/browse/CT-3889)

### Manual testing instructions
- create new OffSar case
- Go to requester details step
- Select Yes
- Should default to Solicitor and relationship text box not present
- Search for address and select one
- When selected company name should be added from address as well
- carry on with entering case as normal
